src/Uno.UI/UI/Xaml/Extensions/FontHelper.Lookup.Android.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Android.App;
+using Android.Graphics;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Uno.Extensions;
+using Uno;
+using Uno.UI;
+using System.Linq;
+using Android.OS;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Text;
+using Uno.Logging;
+using Microsoft.Extensions.Logging;
+using Uno.Collections;
+
+namespace Windows.UI.Xaml
+{
+	internal partial class FontHelper
+	{
+		private readonly static FontFamilyToTypeFaceDictionary _fontFamilyToTypeFaceDictionary = new FontFamilyToTypeFaceDictionary();
+
+		private class FontFamilyToTypeFaceDictionary
+		{
+			public class Entry
+			{
+				private readonly int _hashCode;
+
+				public Entry(FontFamily fontFamily, FontWeight fontWeight, TypefaceStyle style)
+				{
+					FontFamily = fontFamily;
+					FontWeight = fontWeight;
+					Style = style;
+					_hashCode = FontFamily.GetHashCode() ^ FontWeight.GetHashCode();
```

Comment:
Looks strange that hash code doesn't take `Style` into account, while `Equals` does.

I'm also not sure that caching hash code is ideal. `FontFamily` isn't a `sealed` class. If someone derived from it and made a non-immutable class, the cached value can be invalid after a certain operation.

Reply:
Good catch. Indeed, the font family should be the actual string name instead. The `Style` is not there because it's particularly useful in the hash as seemingly it does not change much, so there's not much gain in adding it to the hash at it may not collide much.